ingestion-beam/src/main/java/com/mozilla/telemetry/util/Json.java

```diff
@@ -105,24 +106,48 @@ public static Schema readBigQuerySchema(byte[] data) throws IOException {
   }
 
   /**
-   * Read a {@link JSONObject} from a byte array.
-   *
-   * <p>Using a Jackson {@link ObjectMapper} here is better about not changing long integers to
-   * strings than using {@code new JSONObject(new String(data)}, and accepts {@code byte[]} so that
-   * we don't need to decode to a {@link String} first.
+   * Read an instance of {@code klass} from a byte array.
    *
    * @exception IOException if {@code data} does not contain a valid json object.
    */
-  @SuppressWarnings("checkstyle:AbbreviationAsWordInName")
-  public static JSONObject readJSONObject(byte[] data) throws IOException {
+  public static <T> T readValue(byte[] data, Class<T> klass) throws IOException {
     // Read data into a tree
     TreeNode tree = MAPPER.readTree(data);
     // Check that we have an object, because treeToValue won't
     if (tree == null || !tree.isObject()) {
       throw new IOException("json value is not an object");
     }
-    // Return a JSONObject created from tree
-    return MAPPER.treeToValue(tree, JSONObject.class);
+    // Return a class instance created from the tree
+    return MAPPER.treeToValue(tree, klass);
+  }
+
+  /**
+   * Read bytes into a tree of {@link com.fasterxml.jackson.databind.JsonNode}.
+   *
+   * @exception IOException if {@code data} does not contain a valid json object.
+   */
+  public static ObjectNode readTree(byte[] data) throws IOException {
+    // Read data into a tree
+    TreeNode root = MAPPER.readTree(data);
+    // Check that we have an object, because treeToValue won't
+    if (root == null || !root.isObject()) {
+      throw new IOException("json value is not an object");
+    }
+    return (ObjectNode) root;
+  }
+
+  /**
+   * Return a new, empty {@link ObjectNode}.
+   */
+  public static ObjectNode emptyTree() {
```

Comment:
I would rather these be named `createObjectNode` and `createArrayNode`, like the `MAPPER` methods they wrap
```suggestion
  public static ObjectNode createObjectNode() {
```